lib/internal/modules/esm/loader.js

```diff
@@ -515,9 +522,24 @@ function createModuleLoader(useCustomLoadersIfPresent = true) {
       !require('internal/modules/esm/utils').isLoaderWorker()) {
     const userLoaderPaths = getOptionValue('--experimental-loader');
     if (userLoaderPaths.length > 0) {
-      if (!emittedExperimentalWarning) {
-        emitExperimentalWarning('Custom ESM Loaders');
-        emittedExperimentalWarning = true;
+      if (!emittedLoaderFlagWarning) {
+        const readableURIEncode = (string) => ArrayPrototypeReduce(
+          [
+            [/'/g, '%27'], // We need to URL-encode the single quote as it's the delimiter for the --import flag.
```

Comment:
We need to encode it, or we could just escape it, right? Like this could be `[/'/g, '\\']`?

I don’t really care, I think an apostrophe in a filename is rare enough that it doesn’t matter whether it’s escaped or URL-encoded, but escaping it feels more correct in this case if it’s valid as part of the URL but we can’t use it because it’s the delimiter.

Reply:
The escape character is OS-dependent, e.g. on Windows a backslash doesn’t work

Reply:
I’ve never heard that before. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#string_literals doesn’t mention Windows. Can you point me toward a reference?

Reply:
On PowerShell it’s the <code>`</code> char: https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_special_characters?view=powershell-7.3
On CMD.exe, it’s `^`: https://en.wikipedia.org/wiki/Escape_character#Windows_Command_Prompt

Reply:
But we're inside a single-quoted string here; why would the shell interpreter do anything in here?

Reply:
Nevermind, I get it, the string would terminate prematurely because the single quote isn't escaped per the shell.